### DISCUSSION TITLE
Why the Header files are so complex in fastdds?

### DISCUSSION THREAD

**Discussion by wddpfx** [eProsima/Fast-DDS#5349] [General]
![dds](https://github.com/user-attachments/assets/afb7cb3f-f1f2-4e7c-9dc1-c327c5387c96)

1, there are three folder in the include.But as I know, every file in the "dds" folder is not used. so why do not delete it?

![rtps](https://github.com/user-attachments/assets/023453b6-4fb9-41f7-8c11-9eb23614e759)
2,Why there are so many "rtps"folder, And some file in the include\fastrtps\rtps\builtin just refer the file int the include\fastdds\rtps\builtin

**Comment by JesusPoderoso** [comment]
Hi @wddpfx,
In (latest) Fast DDS v3, the [include](https://github.com/eProsima/Fast-DDS/tree/master/include) directory has been refactored.

**Comment by wddpfx** [comment]
好的，收到了你的邮件，我会尽快查看。
